packages/recs/src/types.ts

```diff
@@ -1,16 +1,14 @@
 import { Subject } from "rxjs";
 import { Type } from "./constants";
-import type { Opaque } from "type-fest";
+import { Opaque } from "type-fest";
 
 /**
- * Used to refer to the index of an entity in a {@link World}.
+ * Entities are represented as symbols internally for memory efficiency.
+ * To get the entity's string representation, use `getEntityString(entitySymbol)`
  */
-export type EntityIndex = Opaque<number, "EntityIndex">;
+export type EntitySymbol = Opaque<symbol, "EntitySymbol">;
 
-/**
- * Used to refer to the string id of an entity (independent from a {@link World}).
- */
-export type EntityID = Opaque<string, "EntityID">;
+export type Entity = Opaque<string, "Entity">;
```

Comment:
is there an advantage of using this custom type instead of just a string? are plain strings compatible with this type?